Sources/Combine/Paging/PageablePresenter.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  PageablePresenter.swift
+//  Catalog
+//
+//  Created by Antonijo Bezmalinovic on 10.01.2023..
+//  Copyright (c) 2021 Infinum. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+@available(iOS 13.0, *)
+protocol PageablePresenter {
+    
+    /// Defines closures for implementator to implement
+    typealias PageableResultClosure = (([Pageable], (any Page)?) -> AnyPublisher<any Page, PagingError>)
+    typealias HasNextPageClosure = (([Pageable], (any Page)?) -> Bool)
+
+    typealias Container = [Pageable]
+    
+    /// Defines interface for setting up pagination
+    func setupPagination(
+        nextPagePublisher: AnyPublisher<Void, Never>,
+        reloadPublisher: AnyPublisher<Void, Never>,
+        nextPage: @escaping PageableResultClosure,
+        hasNextPage: @escaping HasNextPageClosure
+    ) -> AnyPublisher<[Pageable], PagingError>
+}
+
+@available(iOS 13.0, *)
+extension PageablePresenter {
+    
+    /// Default generic pagination implementation
+    func setupPagination(
+        nextPagePublisher: AnyPublisher<Void, Never>,
+        reloadPublisher: AnyPublisher<Void, Never>,
+        nextPage: @escaping PageableResultClosure,
+        hasNextPage: @escaping HasNextPageClosure
+    ) -> AnyPublisher<[Pageable], PagingError> {
+        
+        let shows = page(loadNextPage: nextPagePublisher, reload: reloadPublisher, nextPage: nextPage, hasNextPage: hasNextPage)
+        
+        return shows
+            .compactMap { $0 }
+            .eraseToAnyPublisher()
+    }
+    
+    func page(
+        loadNextPage: AnyPublisher<Void, Never>,
+        reload: AnyPublisher<Void, Never>,
+        nextPage: @escaping PageableResultClosure,
+        hasNextPage: @escaping HasNextPageClosure
+    ) -> AnyPublisher<[Pageable], PagingError> {
```

Comment:
I feel like this should be something that is contained inside the interactor, what do you think @kvaljeva? Maybe we could do something more along the lines of the NetwokPaginationService we did?